packages/sdk-keysigner/index.ts

```diff
@@ -78,6 +78,62 @@ export class BrowserKeySigner implements CredentialSigner<KeyAssertion> {
     ])
   }
 
+  toHex = (buffer: ArrayBuffer): string => {
+    const view = new Uint8Array(buffer)
+    let hexString = ""
+    for (const byte of view) {
+      const hexByte = byte.toString(16)
+      hexString += hexByte.padStart(2, "0")
+    }
+    return hexString.toLowerCase()
+  }
+
+  exportPublicKeyInPemFormat = async (key: CryptoKeyPair): Promise<string> => {
+    const exported = await crypto.subtle.exportKey('spki', key.publicKey)
+    const pem = `-----BEGIN PUBLIC KEY-----\n${toBase64(exported)}\n-----END PUBLIC KEY-----`
+    return pem
+  }
+
+  async create(challenge: UserRegistrationChallenge): Promise<KeyAttestation> {
+    const publicKeyPem = await this.exportPublicKeyInPemFormat(this.options.keyPair)
+    const clientData = JSON.stringify({
+        type: 'key.create',
+        challenge: challenge.challenge,
+        origin: this.options.appOrigin,
+        crossOrigin: this.options.crossOrigin ?? false,
+      })
+    const clientDataHash = await crypto.subtle.digest('SHA-256', new TextEncoder().encode(clientData))
+    const clientDataHashHex = this.toHex(clientDataHash)
+    const credInfoFingerprint = JSON.stringify({
+      clientDataHash: clientDataHashHex,
+      publicKey: publicKeyPem,
+    })
+    let rawSignature: ArrayBuffer
+    const algorithm = this.options.keyPair.privateKey.algorithm.name
+    if (algorithm == 'ECDSA') {
+      rawSignature = await crypto.subtle.sign(
+        { name: 'ECDSA', hash: { name: 'SHA-256' } },
+        this.options.keyPair.privateKey,
+        new TextEncoder().encode(credInfoFingerprint)
+      )
+    } else {
+      throw new Error(`${algorithm} is not supported`)
+    }
+    const signature = this.rawSignatureToAns1(new Uint8Array(rawSignature))
```

Comment:
fyi (not directly linked to this PR) I'm not sure how reliable/accurate the `rawSignatureToAns1` function actually is. I have never digged really. Noah took it from Cody's code I think

Reply:
We could switch to `KJUR/jsrsasign` though I think keeping dependencies light can always be helpful.

Reply:
Yeah we can keep this, it's just that by reading the function, without digging on the actual spec, I cannot tell if the function is actually good or not. I guess empiric proof is good as well, we'll see if some customer complains sometime :)

Reply:
Yes this is pretty straightforward and will not require updates so I'd advocate to keep it in the code without adding an unnecessary dependency.
I tested it the complete flow and was able to go from Registration Code to Registration with Key Credential and Login so that seems to work as expected.

Reply:
I moved this to a utils function